hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -2217,3 +2217,47 @@ def functions(like=lambda: None, returns=none()):
             "but got non-callable like=%r" % (nicerepr(like),)
         )
     return FunctionStrategy(like, returns)
+
+
+@composite
+def slices(draw, size):
+    """Generates slices that will select indices up to the supplied size
+
+    Generated slices will have start and stop indices that range from 0 to size - 1
+    and will step in the appropriate direction. Slices should only produce an empty selection
+    if the start and end are the same.
+
+    Examples from this strategy shrink toward 0 and smaller values
+    """
+    check_valid_integer(size)
+    if size is None or size < 1:
+        raise InvalidArgument("Size must be at least one")
+
+    min_start = min_stop = 0
+    max_start = max_stop = size
+    min_step = 1
+
+    start = draw(one_of(integers(min_start, max_start), none()))  # Start is inclusive
```

Comment:
I find `draw(one_of(integers(min_start, max_start), none()))` easier to read using the `|` operator, as `draw(integers(min_start, max_start) | none())`, since there are fewer nested parentheses.  

We could drop the comments here, or put `# For slices, start is inclusive and stop exclusive` as a line above them.